metadata-ingestion/src/datahub/ingestion/source/data_lake.py

```diff
@@ -0,0 +1,991 @@
+        conf.set("spark.driver.memory", config.spark_driver_memory)
+
+        self.spark = SparkSession.builder.config(conf=conf).getOrCreate()
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = DataLakeSourceConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def read_file(self, file: str) -> Optional[DataFrame]:
+
+        if file.endswith(".parquet"):
+            df = self.spark.read.parquet(file)
+        elif file.endswith(".csv"):
+            # see https://sparkbyexamples.com/pyspark/pyspark-read-csv-file-into-dataframe
+            df = self.spark.read.csv(
+                file,
+                header="True",
+                inferSchema="True",
+                sep=",",
+                ignoreLeadingWhiteSpace=True,
+                ignoreTrailingWhiteSpace=True,
+            )
+        elif file.endswith(".tsv"):
+            df = self.spark.read.csv(
+                file,
+                header="True",
+                inferSchema="True",
+                sep="\t",
+                ignoreLeadingWhiteSpace=True,
+                ignoreTrailingWhiteSpace=True,
+            )
+        elif file.endswith(".json"):
+            df = self.spark.read.json(file)
+        # TODO: add support for more file types
+        # elif file.endswith(".orc"):
+        #     df = self.spark.read.orc(file)
+        # elif file.endswith(".avro"):
+        #     df = self.spark.read.avro(file)
+        else:
+            self.report.report_warning(file, f"file {file} has unsupported extension")
+            return None
+
+        # replace periods in names because they break PyDeequ
+        # see https://mungingdata.com/pyspark/avoid-dots-periods-column-names/
+        return df.toDF(*(c.replace(".", "_") for c in df.columns))
+
+    def get_table_schema(
+        self, dataframe: DataFrame, file_path: str, file_urn_path: str
+    ) -> Iterable[MetadataWorkUnit]:
+
+        datasetUrn = f"urn:li:dataset:(urn:li:dataPlatform:{self.source_config.platform},{file_urn_path},{self.source_config.env})"
+
+        if self.source_config.platform == "s3":
+            datasetUrn = make_s3_urn(file_path, self.source_config.env)
+
+        dataset_snapshot = DatasetSnapshot(
+            urn=datasetUrn,
+            aspects=[],
+        )
+
+        dataset_properties = DatasetPropertiesClass(
+            description="",
+            customProperties={},
+        )
+        dataset_snapshot.aspects.append(dataset_properties)
+
+        column_fields = []
+
+        for field in dataframe.schema.fields:
+
+            field = SchemaField(
+                fieldPath=field.name,
+                type=get_column_type(self.report, "test", field.dataType),
+                nativeDataType=str(field.dataType),
+                recursive=False,
+            )
+
+            column_fields.append(field)
+
+        schema_metadata = SchemaMetadata(
+            schemaName="test",
```

Comment:
does schemaName="test" mean something special here?

Reply:
Nah – I forgot to rewrite it when testing. Thanks for the catch!